js/web/lib/wasm/jsep/webgpu/ops/split.ts

```diff
@@ -107,7 +107,7 @@ const createSplitProgramInfo = (inputs: readonly TensorView[], attributes: Split
   }`;
   return {
     name: 'Split',
-    shaderCache: {hint: attributes.cacheKey, inputDependencies: ['rank']},
+    shaderCache: {hint: `${attributes.cacheKey};${inputShape}`, inputDependencies: ['rank']},
```

Comment:
Do you still need this fix after #19554 is merged?

Reply:
And anyway this line is a workaround.

If you want to put input shape into shaderCache, you should use inputDependencies set to 'dims', and don't use input shapes into uniforms.

Reply:
Reverted the changes to Split.